In [ ]:
import pandas as pd
import numpy as np

In [ ]:
data_api = pd.read_csv('zoon_data_api_full.csv')
data_api.head()

Название  Группа найдена  \
0  Оздоровительный центр на Соколе               1   
1   Андреевские больницы - НЕБОЛИТ               1   
2                          WeiDent               0   
3          Клиника на Потаповскомъ               1   
4              Юсуповская больница               0   

               Тематика группы  Число подписчиков  Число постов за 6 месяцев  \
0  Государственная организация               1598                         99   
1            Медицинский центр                511                         85   
2                   Неизвестно                  0                          0   
3            Медицинский центр                 16                          0   
4                   Неизвестно                  0                          0   

   Среднее число комментариев на пост  Среднее число лайков на пост  \
0                                0.08                          8.49   
1                                0.05                          2.13   
2                                0.00                          0.00   
3                                0.00                          0.00   
4                                0.00                          0.00   

   Среднее число просмотров на пост  Процент вовлеченности (ER)  
0                            712.53                       53.13  
1                            110.76                       36.20  
2                              0.00                        0.00  
3                              0.00                        0.00  
4                              0.00                        0.00

In [ ]:
data_all = pd.read_csv('zoon_data_all_full_.csv')
data_all.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                              Ссылка  \
0  https://zoon.ru/msk/medical/ozdorovitelnyj_tse...   
1  https://zoon.ru/msk/medical/meditsinskij_tsent...   
2  https://zoon.ru/msk/medical/stomatologicheskay...   
3  https://zoon.ru/msk/medical/klinika_na_potapov...   
4  https://zoon.ru/msk/medical/yusupovskaya_bolni...   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

  Время до метро  Рейтинг Отзывы  \
0         9 мин.      5.0     37   
1        13 мин.      4.8    141   
2        21 мин.      5.0    290   
3         9 мин.      4.9    156   
4         8 мин.      4.8    477   

                                        Подкатегории  
0  Расслабляющий массаж, Массаж спины, Лимфодрена...  
1  Фракционный фототермолиз, Инъекции коллагена, ...  
2  Стоматология под микроскопом, Базальная имплан...  
3  Фракционный фототермолиз, Онкологический диспа...  
4  Онкологический диспансер, Перевозка больных, В...

Мы загрузили оба датсета. Теперь мы их соединим (по названию) и сделаем EDA-анализ.

Сначал обработаем датасет 'data_all'. В его анализ войдут следующие пункты:

1) Выделение из адреса города и этажа (где возможно)

2) Выделение значений типа int в столбце "Время до метро" (также предобработка значений "Нет данных" и тех значений, которые указаны в километрах)

3) Формирование нового признака "Релевантная оценка" на основе столбцов "Рейтинг" и "Отзывы"(количество отзывов)

4)Выбор из подкатегории наиболее релевантной категории и занесение ее в отдельный столбец




Затем обработаем признаки из датасета "data_api". Сделаем следующие шаги:

In [ ]:
merged_df = data_all.merge(data_api, left_on = "Название", right_on = "Название")
merged_df.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                              Ссылка  \
0  https://zoon.ru/msk/medical/ozdorovitelnyj_tse...   
1  https://zoon.ru/msk/medical/meditsinskij_tsent...   
2  https://zoon.ru/msk/medical/stomatologicheskay...   
3  https://zoon.ru/msk/medical/klinika_na_potapov...   
4  https://zoon.ru/msk/medical/yusupovskaya_bolni...   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

  Время до метро  Рейтинг Отзывы  \
0         9 мин.      5.0     37   
1        13 мин.      4.8    141   
2        21 мин.      5.0    290   
3         9 мин.      4.9    156   
4         8 мин.      4.8    477   

                                        Подкатегории  Группа найдена  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...               1   
1  Фракционный фототермолиз, Инъекции коллагена, ...               1   
2  Стоматология под микроскопом, Базальная имплан...               0   
3  Фракционный фототермолиз, Онкологический диспа...               1   
4  Онкологический диспансер, Перевозка больных, В...               0   

               Тематика группы  Число подписчиков  Число постов за 6 месяцев  \
0  Государственная организация               1598                         99   
1            Медицинский центр                511                         85   
2                   Неизвестно                  0                          0   
3            Медицинский центр                 16                          0   
4                   Неизвестно                  0                          0   

   Среднее число комментариев на пост  Среднее число лайков на пост  \
0                                0.08                          8.49   
1                                0.05                          2.13   
2                                0.00                          0.00   
3                                0.00                          0.00   
4                                0.00                          0.00   

   Среднее число просмотров на пост  Процент вовлеченности (ER)  
0                            712.53                       53.13  
1                            110.76                       36.20  
2                              0.00                        0.00  
3                              0.00                        0.00  
4                              0.00                        0.00

In [ ]:
merged_df['Название'].value_counts()

Название
Планета здоровья      55
Столички              47
DDX Fitness           42
MIUZ Diamonds         40
Очкарик               29
                      ..
La venus               1
Братья Чистовы         1
У Анны                 1
У Риммы                1
Петухов и партнеры     1
Name: count, Length: 8124, dtype: int64

In [ ]:
file_path = 'merged.csv'
merged_df.to_csv(file_path, index=False, encoding='UTF-8-SIG')

Заметим, что хоть у нас и 10126 строк, уникальных организаций у нас 8124. Это связано с тем, что если организация сетевая, то каждая сеть заводит свой аккаунт в соцсетях. таким образом, мы можем наблюдать, что, например, у сети "Планета здоровья" 55 точек, у которых есть аккаунт в ВК, у "Столички" - 47 и так далее. Либо жто просто множество созданных аккаунтов у какой-либо аптеки, магазина и т. д.

In [ ]:


def extract_city_and_metro(address):
    parts = address.split(',')
    city = parts[0].strip() if len(parts) > 0 else None
    metro = parts[-1].strip() if len(parts) > 1 else None
    return pd.Series([city, metro])

new_columns = merged_df['Адрес места'].apply(extract_city_and_metro)
merged_df[['Город', 'Метро']] = new_columns

merged_df.head()


AttributeError: 'float' object has no attribute 'split'

In [ ]:
import re

def extract_city_and_floor(address):
    parts = address.split(',')
    city = parts[-1].strip() if len(parts) > 1 else None
    floor_match = re.search(r'([0-9]+)', address)
    floor = floor_match.group(0) if floor_match else None
    return pd.Series([city, floor])

new_columns = cleaned_df['Адрес места'].apply(extract_city_and_floor)
cleaned_df[['Город', 'Этаж']] = new_columns

cleaned_df.head()

updated_file_path_with_floor = 'zoon_data_with_city_and_floor.csv'
cleaned_df.to_csv(updated_file_path_with_floor, index=False, encoding='UTF-8-SIG')


                          Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      13.0      4.8   141.0   
2                      21.0      5.0   290.0   
3                       9.0      4.9   156.0   
4                       8.0      4.8   477.0   

                                        Подкатегории        Г

In [ ]:
cleaned_df = pd.read_csv('zoon_data_last_phrase_category.csv')

In [ ]:
def extract_last_phrase(category):
    if isinstance(category, str):
        return category.split(',')[-1].strip()
    return None

cleaned_df['Категория'] = cleaned_df['Категория'].apply(extract_last_phrase)

cleaned_df.head()



                          Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                              Ссылка  \
0  https://zoon.ru/msk/medical/ozdorovitelnyj_tse...   
1  https://zoon.ru/msk/medical/meditsinskij_tsent...   
2  https://zoon.ru/msk/medical/stomatologicheskay...   
3  https://zoon.ru/msk/medical/klinika_na_potapov...   
4  https://zoon.ru/msk/medical/yusupovskaya_bolni...   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

  Время до ме

In [ ]:
cleaned_df['Категория'] = cleaned_df['Категория'].replace('₽', 'Рестораны')


In [ ]:
cleaned_df['Категория'] = cleaned_df['Категория'].replace('•', np.nan)
cleaned_df['Категория'] = cleaned_df['Категория'].fillna(method='ffill')


<ipython-input-130-24e58de3e152>:2: FutureWarning: Series.fillna with 'method' is deprecated and will raise in a future version. Use obj.ffill() or obj.bfill() instead.
  cleaned_df['Категория'] = cleaned_df['Категория'].fillna(method='ffill')


In [ ]:
cleaned_df[cleaned_df['Категория'] == '•']

Empty DataFrame
Columns: [Название, Ссылка, Адрес места, Метро, Время до метро, Рейтинг, Отзывы, Подкатегории, Город, Этаж, Категория]
Index: []

In [ ]:
updated_file_path_last_phrase = 'zoon_data_last_phrase_category.csv'
cleaned_df.to_csv(updated_file_path_last_phrase, index=False, encoding='UTF-8-SIG')

print('Updated dataframe with last phrase in category saved to:', updated_file_path_last_phrase)

Updated dataframe with last phrase in category saved to: zoon_data_last_phrase_category.csv


Удалить ссылку, отделить этаж, город и адрес, убрать мин из метро и записать их в название столбца, сформировать новый признак на основе оценки и количестве пользователей, делаем сплит по подкатегориям,

In [ ]:
cleaned_df = cleaned_df.drop(['Ссылка'], axis=1)

In [ ]:
unique_time_values = cleaned_df['Время до метро'].unique()
print(unique_time_values)

['9 мин.' '13 мин.' '21 мин.' '8 мин.' 'Нет данных' '7 мин.' '18 мин.'
 '10 мин.' '4 мин.' '5 мин.' '2 мин.' '14 мин.' '26 мин.' '3 мин.'
 '17 мин.' '12 мин.' '6 мин.' '1 мин.' '4.2 км' '49 мин.' '43 мин.'
 '30 мин.' '4.7 км' '22 мин.' '23 мин.' '4.8 км' '24 мин.' '16 мин.'
 '20 мин.' '11 мин.' '42 мин.' '19 мин.' '25 мин.' '28 мин.' '27 мин.'
 '4.3 км' '29 мин.' '55 мин.' '36 мин.' '53 мин.' '15 мин.' '35 мин.'
 '44 мин.' '41 мин.' '4.5 км' '38 мин.' '2.9 км' '58 мин.' '37 мин.'
 '45 мин.' '34 мин.' '4.6 км' '56 мин.' '57 мин.' '33 мин.' '59 мин.'
 '3.1 км' '46 мин.' '32 мин.' '4.1 км' '39 мин.' '4.4 км' '52 мин.'
 '48 мин.' '40 мин.' '50 мин.' '2.5 км' '4.9 км' '5 км' '51 мин.'
 '31 мин.' '47 мин.' '54 мин.' '5.3 км' '1535 м' '5.8 км' '3.7 км'
 '2.2 км' '760 м' '2.3 км' '28.3 км' '3.8 км' '3.5 км' '654 м' '15.6 км'
 '24.5 км' '1384 м' '154 м' '531 м' '6.2 км']


In [ ]:
def clean_time(value):
    if isinstance(value, str):
        if 'мин.' in value:
            return float(value.replace('мин.', '').strip())
        elif 'км' in value:
            return np.nan
        elif value == 'Нет данных':
            return np.nan
    return np.nan

cleaned_df['Время до метро'] = cleaned_df['Время до метро'].apply(clean_time)

cleaned_df.rename(columns={'Время до метро': 'Время до метро в минутах'}, inplace=True)



In [ ]:
cleaned_df.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг Отзывы  \
0                       9.0      5.0     37   
1                      13.0      4.8    141   
2                      21.0      5.0    290   
3                       9.0      4.9    156   
4                       8.0      4.8    477   

                                        Подкатегории        Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1  Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2  Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3  Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4  Онкологический диспансер, Перевозка больных, В...       Москва   NaN   

            Категория  
0  Медицинские центры  
1  Медицинские центры  
2  Медицинские центры  
3  Медицинские центры  
4  Медицинские центры

In [ ]:
cleaned_df['Отзывы'] = pd.to_numeric(cleaned_df['Отзывы'], errors='coerce')


cleaned_df['Релевантная оценка'] = cleaned_df['Рейтинг'] * cleaned_df['Отзывы']

updated_file_path_final = 'zoon_data_final.csv'
cleaned_df.to_csv(updated_file_path_final, index=False, encoding='UTF-8-SIG')

cleaned_df.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      13.0      4.8   141.0   
2                      21.0      5.0   290.0   
3                       9.0      4.9   156.0   
4                       8.0      4.8   477.0   

                                        Подкатегории        Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1  Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2  Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3  Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4  Онкологический диспансер, Перевозка больных, В...       Москва   NaN   

            Категория  Релевантная оценка  
0  Медицинские центры               185.0  
1  Медицинские центры               676.8  
2  Медицинские центры              1450.0  
3  Медицинские центры               764.4  
4  Медицинские центры              2289.6

In [ ]:
cleaned_df = pd.read_csv('zoon_data_final.csv')
cleaned_df.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      13.0      4.8   141.0   
2                      21.0      5.0   290.0   
3                       9.0      4.9   156.0   
4                       8.0      4.8   477.0   

                                        Подкатегории        Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1  Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2  Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3  Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4  Онкологический диспансер, Перевозка больных, В...       Москва   NaN   

            Категория  Релевантная оценка  
0  Медицинские центры               185.0  
1  Медицинские центры               676.8  
2  Медицинские центры              1450.0  
3  Медицинские центры               764.4  
4  Медицинские центры              2289.6

In [ ]:
cleaned_df = cleaned_df.merge(data_api, left_on = "Название", right_on = "Название")
cleaned_df.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      13.0      4.8   141.0   
2                      21.0      5.0   290.0   
3                       9.0      4.9   156.0   
4                       8.0      4.8   477.0   

                                        Подкатегории        Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1  Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2  Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3  Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4  Онкологический диспансер, Перевозка больных, В...       Москва   NaN   

            Категория  Релевантная оценка  Группа найдена  \
0  Медицинские центры               185.0               1   
1  Медицинские центры               676.8               1   
2  Медицинские центры              1450.0               0   
3  Медицинские центры               764.4               1   
4  Медицинские центры              2289.6               0   

               Тематика группы  Число подписчиков  Число постов за 6 месяцев  \
0  Государственная организация               1598                         99   
1            Медицинский центр                511                         85   
2                   Неизвестно                  0                          0   
3            Медицинский центр                 16                          0   
4                   Неизвестно                  0                          0   

   Среднее число комментариев на пост  Среднее число лайков на пост  \
0                                0.08                          8.49   
1                                0.05                          2.13   
2                                0.00                          0.00   
3                                0.00                          0.00   
4                                0.00                          0.00   

   Среднее число просмотров на пост  Процент вовлеченности (ER)  
0                            712.53                       53.13  
1                            110.76                       36.20  
2                              0.00                        0.00  
3                              0.00                        0.00  
4                              0.00                        0.00

In [ ]:
cleaned_df['Среднее число комментариев на пост'] = cleaned_df['Среднее число комментариев на пост'] * 100

In [ ]:
cleaned_df.rename(columns={'Среднее число комментариев на пост': 'Среднее число комментариев на пост(%)'}, inplace=True)
cleaned_df.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      13.0      4.8   141.0   
2                      21.0      5.0   290.0   
3                       9.0      4.9   156.0   
4                       8.0      4.8   477.0   

                                        Подкатегории        Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1  Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2  Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3  Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4  Онкологический диспансер, Перевозка больных, В...       Москва   NaN   

            Категория  Релевантная оценка  Группа найдена  \
0  Медицинские центры               185.0               1   
1  Медицинские центры               676.8               1   
2  Медицинские центры              1450.0               0   
3  Медицинские центры               764.4               1   
4  Медицинские центры              2289.6               0   

               Тематика группы  Число подписчиков  Число постов за 6 месяцев  \
0  Государственная организация               1598                         99   
1            Медицинский центр                511                         85   
2                   Неизвестно                  0                          0   
3            Медицинский центр                 16                          0   
4                   Неизвестно                  0                          0   

   Среднее число комментариев на пост(%)  Среднее число лайков на пост  \
0                                    8.0                          8.49   
1                                    5.0                          2.13   
2                                    0.0                          0.00   
3                                    0.0                          0.00   
4                                    0.0                          0.00   

   Среднее число просмотров на пост  Процент вовлеченности (ER)  
0                            712.53                       53.13  
1                            110.76                       36.20  
2                              0.00                        0.00  
3                              0.00                        0.00  
4                              0.00                        0.00

In [ ]:
updated_file_path_final = 'zoon_data_final.csv'
cleaned_df.to_csv(updated_file_path_final, index=False, encoding='UTF-8-SIG')

In [ ]:
import pandas as pd
import numpy as np
data = pd.read_csv('zoon_data_final-2.csv')

In [ ]:
data['Категория'].value_counts()

Категория
Для детей                  301
Детские сады               301
Одежда и обувь             300
Госуслуги                  300
Квесты                     300
Культурные места           300
Юридические услуги         300
Туризм                     300
Финансы                    300
Недвижимость               300
Ритуальные услуги          299
Строительство              299
Образование                299
Сервисные центры           299
Магазины                   299
Торговые центры            299
Йога центры                299
Интернет компании          299
Дома отдыха                299
Услуги для бизнеса         299
Бытовые услуги             299
Развлекательные центры     299
Курсы                      299
Рестораны                  299
Оптики                     298
Аптеки                     297
Сауны и бани               297
Фитнес клубы               297
Автосервисы                294
Салоны красоты и СПА       292
Ветеринарные клиники       291
Медицинские центры         290
Ночные клубы               273
Кинотеатры                 253
Студии красоты               5
Медцентр                     3
Мед центр                    3
Ветклиника                   3
Ремонт автомашины            1
Частная клиника              1
Медклиника                   1
Частная поликлиника          1
Официальный автосервис       1
Ветлечебница                 1
Авторемонт                   1
Автомастерская               1
Ремонт авто                  1
Популярные рестораны         1
Аптечные пункты              1
Аптечные киоски              1
Аптечные магазины            1
Спортклуб                    1
(МЦ) медицинские центры      1
Зооклиника                   1
Ветеринарка                  1
Снять сауну                  1
Ремонт автомобилей           1
Name: count, dtype: int64

In [ ]:
data['Категория'] = data['Категория'].replace('Студии красоты', 'Салоны красоты и СПА')
data['Категория'] = data['Категория'].replace('Медцентр', 'Медицинские центры')
data['Категория'] = data['Категория'].replace('Мед центр', 'Медицинские центры')
data['Категория'] = data['Категория'].replace('Ветклиника', 'Ветеринарные клиники')

data['Категория'] = data['Категория'].replace('Ремонт автомашины', 'Автосервисы')
data['Категория'] = data['Категория'].replace('Частная клиника', 'Медицинские центры')
data['Категория'] = data['Категория'].replace('Частная поликлиника', 'Медицинские центры')
data['Категория'] = data['Категория'].replace('Медклиника', 'Медицинские центры')
data['Категория'] = data['Категория'].replace('Официальный автосервис', 'Автосервисы')
data['Категория'] = data['Категория'].replace('Ветлечебница', 'Ветеринарные клиники')
data['Категория'] = data['Категория'].replace('Авторемонт', 'Автосервисы')
data['Категория'] = data['Категория'].replace('Автомастерская', 'Автосервисы')

data['Категория'] = data['Категория'].replace('Ремонт авто', 'Автосервисы')
data['Категория'] = data['Категория'].replace('Популярные рестораны', 'Рестораны')
data['Категория'] = data['Категория'].replace('Аптечные пункты', 'Аптеки')
data['Категория'] = data['Категория'].replace('Аптечные киоски', 'Аптеки')
data['Категория'] = data['Категория'].replace('Аптечные магазины', 'Аптеки')

data['Категория'] = data['Категория'].replace('Спортклуб', 'Фитнес клубы')
data['Категория'] = data['Категория'].replace('(МЦ) медицинские центры', 'Медицинские центры')
data['Категория'] = data['Категория'].replace('Зооклиника', 'Ветеринарные клиники')
data['Категория'] = data['Категория'].replace('Ветеринарка', 'Ветеринарные клиники')
data['Категория'] = data['Категория'].replace('Снять сауну', 'Сауны и бани')
data['Категория'] = data['Категория'].replace('Ремонт автомобилей', 'Автосервисы')



In [ ]:
data['Категория'].value_counts()

Категория
Для детей                 301
Детские сады              301
Медицинские центры        300
Аптеки                    300
Финансы                   300
Туризм                    300
Одежда и обувь            300
Недвижимость              300
Культурные места          300
Квесты                    300
Госуслуги                 300
Юридические услуги        300
Автосервисы               300
Рестораны                 300
Бытовые услуги            299
Образование               299
Услуги для бизнеса        299
Курсы                     299
Торговые центры           299
Строительство             299
Ритуальные услуги         299
Йога центры               299
Магазины                  299
Сервисные центры          299
Интернет компании         299
Дома отдыха               299
Развлекательные центры    299
Фитнес клубы              298
Сауны и бани              298
Оптики                    298
Ветеринарные клиники      297
Салоны красоты и СПА      297
Ночные клубы              273
Кинотеатры                253
Name: count, dtype: int64

In [ ]:
data = data.drop(['Релевантная оценка'], axis=1)
data.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      13.0      4.8   141.0   
2                      21.0      5.0   290.0   
3                       9.0      4.9   156.0   
4                       8.0      4.8   477.0   

                                        Подкатегории        Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1  Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2  Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3  Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4  Онкологический диспансер, Перевозка больных, В...       Москва   NaN   

            Категория  Группа найдена              Тематика группы  \
0  Медицинские центры               1  Государственная организация   
1  Медицинские центры               1            Медицинский центр   
2  Медицинские центры               0                   Неизвестно   
3  Медицинские центры               1            Медицинский центр   
4  Медицинские центры               0                   Неизвестно   

   Число подписчиков  Число постов за 6 месяцев  \
0               1598                         99   
1                511                         85   
2                  0                          0   
3                 16                          0   
4                  0                          0   

   Среднее число комментариев на пост(%)  Среднее число лайков на пост  \
0                                    8.0                          8.49   
1                                    5.0                          2.13   
2                                    0.0                          0.00   
3                                    0.0                          0.00   
4                                    0.0                          0.00   

   Среднее число просмотров на пост  Процент вовлеченности (ER)  
0                            712.53                       53.13  
1                            110.76                       36.20  
2                              0.00                        0.00  
3                              0.00                        0.00  
4                              0.00                        0.00

In [ ]:
updated_file_path_final = 'zoon_data_final.csv'
data.to_csv(updated_file_path_final, index=False, encoding='UTF-8-SIG')

In [ ]:
data = pd.read_csv('zoon_data_final-2.csv')
data.head()

Название  \
0  Оздоровительный центр на Соколе   
1   Андреевские больницы - НЕБОЛИТ   
2                          WeiDent   
3          Клиника на Потаповскомъ   
4              Юсуповская больница   

                                         Адрес места         Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва         Сокол   
1                   Спасская улица, 1к3, Красногорск      Мякинино   
2              улица Академика Опарина, 4 к1, Москва       Беляево   
3    Потаповский переулок, 10, вход со двора, Москва  Чистые пруды   
4                Нагорная улица, 17 корпус 6, Москва      Нагорная   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      13.0      4.8   141.0   
2                      21.0      5.0   290.0   
3                       9.0      4.9   156.0   
4                       8.0      4.8   477.0   

                                        Подкатегории        Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1  Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2  Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3  Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4  Онкологический диспансер, Перевозка больных, В...       Москва   NaN   

            Категория  Релевантная оценка  Группа найдена  \
0  Медицинские центры               185.0               1   
1  Медицинские центры               676.8               1   
2  Медицинские центры              1450.0               0   
3  Медицинские центры               764.4               1   
4  Медицинские центры              2289.6               0   

               Тематика группы  Число подписчиков  Число постов за 6 месяцев  \
0  Государственная организация               1598                         99   
1            Медицинский центр                511                         85   
2                   Неизвестно                  0                          0   
3            Медицинский центр                 16                          0   
4                   Неизвестно                  0                          0   

   Среднее число комментариев на пост(%)  Среднее число лайков на пост  \
0                                    8.0                          8.49   
1                                    5.0                          2.13   
2                                    0.0                          0.00   
3                                    0.0                          0.00   
4                                    0.0                          0.00   

   Среднее число просмотров на пост  Процент вовлеченности (ER)  
0                            712.53                       53.13  
1                            110.76                       36.20  
2                              0.00                        0.00  
3                              0.00                        0.00  
4                              0.00                        0.00

In [ ]:
metro = pd.read_excel('Метро.xlsx')
metro.head()

Метро                                    Район
0         Авиамоторная         Восточный Административный Округ
1        Автозаводская             Южный Административный Округ
2        Академическая      Юго-западный Административный Округ
3  Александровский сад       Центральный Административный Округ
4         Алексеевская  Северо-восточный Административный Округ

In [ ]:
merged = data.merge(metro, left_on = "Метро", right_on = "Метро")
merged.head()

Название  \
0  Оздоровительный центр на Соколе   
1                          WeiDent   
2          Клиника на Потаповскомъ   
3              Юсуповская больница   
4                Счастливый Доктор   

                                         Адрес места            Метро  \
0  Балтийская улица, 15, 432 кабинет; 4 этаж, Москва            Сокол   
1              улица Академика Опарина, 4 к1, Москва          Беляево   
2    Потаповский переулок, 10, вход со двора, Москва     Чистые пруды   
3                Нагорная улица, 17 корпус 6, Москва         Нагорная   
4                улица Адмирала Лазарева, 43, Москва  Бунинская аллея   

   Время до метро в минутах  Рейтинг  Отзывы  \
0                       9.0      5.0    37.0   
1                      21.0      5.0   290.0   
2                       9.0      4.9   156.0   
3                       8.0      4.8   477.0   
4                       7.0      4.9   497.0   

                                        Подкатегории   Город  Этаж  \
0  Расслабляющий массаж, Массаж спины, Лимфодрена...  Москва   NaN   
1  Стоматология под микроскопом, Базальная имплан...  Москва   NaN   
2  Фракционный фототермолиз, Онкологический диспа...  Москва   NaN   
3  Онкологический диспансер, Перевозка больных, В...  Москва   NaN   
4  Базальная имплантация, ЭКГ на дому, Прессотера...  Москва   NaN   

            Категория  Релевантная оценка  Группа найдена  \
0  Медицинские центры               185.0               1   
1  Медицинские центры              1450.0               0   
2  Медицинские центры               764.4               1   
3  Медицинские центры              2289.6               0   
4  Медицинские центры              2435.3               1   

               Тематика группы  Число подписчиков  Число постов за 6 месяцев  \
0  Государственная организация               1598                         99   
1                   Неизвестно                  0                          0   
2            Медицинский центр                 16                          0   
3                   Неизвестно                  0                          0   
4    Домашние и дикие животные                377                          3   

   Среднее число комментариев на пост(%)  Среднее число лайков на пост  \
0                                    8.0                          8.49   
1                                    0.0                          0.00   
2                                    0.0                          0.00   
3                                    0.0                          0.00   
4                                  100.0                          9.33   

   Среднее число просмотров на пост  Процент вовлеченности (ER)  \
0                            712.53                       53.13   
1                              0.00                        0.00   
2                              0.00                        0.00   
3                              0.00                        0.00   
4                            887.00                        8.22   

                                 Район  
0      Северный Административный Округ  
1  Юго-западный Административный Округ  
2   Центральный Административный Округ  
3         Южный Административный Округ  
4  Юго-западный Административный Округ

In [ ]:
merged[merged['Район'].isnull() == True]

Empty DataFrame
Columns: [Название, Адрес места, Метро, Время до метро в минутах, Рейтинг, Отзывы, Подкатегории, Город, Этаж, Категория, Релевантная оценка, Группа найдена, Тематика группы, Число подписчиков, Число постов за 6 месяцев, Среднее число комментариев на пост(%), Среднее число лайков на пост, Среднее число просмотров на пост, Процент вовлеченности (ER), Район]
Index: []

In [ ]:
updated_file_path_final = 'zoon_data_final-2.csv'
merged.to_csv(updated_file_path_final, index=False, encoding='UTF-8-SIG')

In [1]:
import pandas as pd

In [24]:
final = pd.read_csv('zoon_data_final (2).csv')
final

Название  \
0      Оздоровительный центр на Соколе   
1       Андреевские больницы - НЕБОЛИТ   
2                              WeiDent   
3              Клиника на Потаповскомъ   
4                  Юсуповская больница   
...                                ...   
10098                  Licenziataxi.ru   
10099                           Фемида   
10100  Листратенко Татьяна Анатольевна   
10101                            Базис   
10102               Петухов и партнеры   

                                             Адрес места             Метро  \
0      Балтийская улица, 15, 432 кабинет; 4 этаж, Москва             Сокол   
1                       Спасская улица, 1к3, Красногорск          Мякинино   
2                  улица Академика Опарина, 4 к1, Москва           Беляево   
3        Потаповский переулок, 10, вход со двора, Москва      Чистые пруды   
4                    Нагорная улица, 17 корпус 6, Москва          Нагорная   
...                                                  ...               ...   
10098     проезд Черепановых, 3, 10 офис; 2 этаж, Москва           Коптево   
10099  площадь Журавлева, 10 ст1, 120 офис; 1 этаж, М...  Электрозаводская   
10100           Люблинская улица, 179/1, 10 этаж, Москва           Марьино   
10101            Пресненская набережная, 10блокС, Москва     Международная   
10102  Пресненский район, 123112, 1-й Красногвардейск...     Международная   

       Время до метро в минутах  Рейтинг  Отзывы  \
0                           9.0      5.0    37.0   
1                          13.0      4.8   141.0   
2                          21.0      5.0   290.0   
3                           9.0      4.9   156.0   
4                           8.0      4.8   477.0   
...                         ...      ...     ...   
10098                       4.0      4.2    12.0   
10099                       2.0      4.3     2.0   
10100                      10.0      2.8    13.0   
10101                       2.0      4.4    13.0   
10102                       5.0      5.0    16.0   

                                            Подкатегории        Город  Этаж  \
0      Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1      Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2      Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3      Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4      Онкологический диспансер, Перевозка больных, В...       Москва   NaN   
...                                                  ...          ...   ...   
10098  Регистрация юридических лиц и предпринимателей...       Москва   NaN   
10099  Юридическое сопровождение бизнеса, Юридическая...       Москва   NaN   
10100  Урегулирование договорных споров, Определение ...       Москва  10.0   
10101  Перевод в нежилое помещение, Юридическая консу...       Москва   NaN   
10102  Составление жалоб, Налоговые консультации, Адв...       Москва   NaN   

                Категория  Группа найдена              Тематика группы  \
0      Медицинские центры               1  Государственная организация   
1      Медицинские центры               1            Медицинский центр   
2      Медицинские центры               0                   Неизвестно   
3      Медицинские центры               1            Медицинский центр   
4      Медицинские центры               0                   Неизвестно   
...                   ...             ...                          ...   
10098  Юридические услуги               0                   Неизвестно   
10099  Юридические услуги               1           Юридические услуги   
10100  Юридические услуги               0                   Неизвестно   
10101  Юридические услуги               1      Программное обеспечение   
10102  Юридические услуги               1           Юридические услуги   

       Число подписчиков  Число постов за 6 месяцев  \
0                   1598                         99   
1              

In [25]:
api_new = pd.read_csv('zoon_data_api_full_.csv')
api_new

Название  Группа найдена  Подписчики скрыты  \
0     Оздоровительный центр на Соколе               1                  0   
1      Андреевские больницы - НЕБОЛИТ               1                  1   
2                             WeiDent               0                  0   
3             Клиника на Потаповскомъ               1                  0   
4                 Юсуповская больница               0                  0   
...                               ...             ...                ...   
8119                         Я Вправе               1                  0   
8120                  Licenziataxi.ru               0                  0   
8121  Листратенко Татьяна Анатольевна               0                  0   
8122                            Базис               1                  0   
8123               Петухов и партнеры               1                  0   

                  Тематика группы  Число подписчиков  \
0     Государственная организация               1600   
1               Медицинский центр                511   
2                      Неизвестно                  0   
3               Медицинский центр                 16   
4                      Неизвестно                  0   
...                           ...                ...   
8119                        Наука                468   
8120                   Неизвестно                  0   
8121                   Неизвестно                  0   
8122      Программное обеспечение               2982   
8123           Юридические услуги                 73   

      Среднее число упоминаний в новостях в день  Процент мужчин  \
0                                           6.83           21.30   
1                                           0.57            0.00   
2                                           0.00            0.00   
3                                           0.00            6.25   
4                                           0.00            0.00   
...                                          ...             ...   
8119                                       33.33           39.96   
8120                                        0.00            0.00   
8121                                        0.00            0.00   
8122                                       33.33           74.10   
8123                                       14.30           45.21   

      Процент женщин  Число постов за 6 месяцев  \
0              78.70                         99   
1               0.00                         86   
2               0.00                          0   
3              93.75                          0   
4               0.00                          0   
...              ...                        ...   
8119           60.04                        100   
8120            0.00                          0   
8121            0.00                          0   
8122           25.90                        100   
8123           54.79                         30   

      Среднее число комментариев на пост  Среднее число лайков на пост  \
0                                   0.08                          8.48   
1                                   0.05                          2.12   
2                                   0.00                          0.00   
3                                   0.00                          0.00   
4                                   0.00                          0.00   
...                                  ...                           ...   
8119                                2.47                         16.35   
8120                                0.00                          0.00   
8121                                0.00                          0.00   
8122                                0.20                         10.06   
8123                                0.13                         11.03   

      Среднее число просмотров на пост  Процент вовлеченности (ER)  
0                               711.48                 

In [26]:
new = api_new[['Название', 'Подписчики скрыты', 'Среднее число упоминаний в новостях в день', 'Процент мужчин', 'Процент женщин']].copy()
new

Название  Подписчики скрыты  \
0     Оздоровительный центр на Соколе                  0   
1      Андреевские больницы - НЕБОЛИТ                  1   
2                             WeiDent                  0   
3             Клиника на Потаповскомъ                  0   
4                 Юсуповская больница                  0   
...                               ...                ...   
8119                         Я Вправе                  0   
8120                  Licenziataxi.ru                  0   
8121  Листратенко Татьяна Анатольевна                  0   
8122                            Базис                  0   
8123               Петухов и партнеры                  0   

      Среднее число упоминаний в новостях в день  Процент мужчин  \
0                                           6.83           21.30   
1                                           0.57            0.00   
2                                           0.00            0.00   
3                                           0.00            6.25   
4                                           0.00            0.00   
...                                          ...             ...   
8119                                       33.33           39.96   
8120                                        0.00            0.00   
8121                                        0.00            0.00   
8122                                       33.33           74.10   
8123                                       14.30           45.21   

      Процент женщин  
0              78.70  
1               0.00  
2               0.00  
3              93.75  
4               0.00  
...              ...  
8119           60.04  
8120            0.00  
8121            0.00  
8122           25.90  
8123           54.79  

[8124 rows x 5 columns]

In [27]:
final = final.merge(new, left_on = "Название", right_on = "Название")
final


Название  \
0      Оздоровительный центр на Соколе   
1       Андреевские больницы - НЕБОЛИТ   
2                              WeiDent   
3              Клиника на Потаповскомъ   
4                  Юсуповская больница   
...                                ...   
10098                  Licenziataxi.ru   
10099                           Фемида   
10100  Листратенко Татьяна Анатольевна   
10101                            Базис   
10102               Петухов и партнеры   

                                             Адрес места             Метро  \
0      Балтийская улица, 15, 432 кабинет; 4 этаж, Москва             Сокол   
1                       Спасская улица, 1к3, Красногорск          Мякинино   
2                  улица Академика Опарина, 4 к1, Москва           Беляево   
3        Потаповский переулок, 10, вход со двора, Москва      Чистые пруды   
4                    Нагорная улица, 17 корпус 6, Москва          Нагорная   
...                                                  ...               ...   
10098     проезд Черепановых, 3, 10 офис; 2 этаж, Москва           Коптево   
10099  площадь Журавлева, 10 ст1, 120 офис; 1 этаж, М...  Электрозаводская   
10100           Люблинская улица, 179/1, 10 этаж, Москва           Марьино   
10101            Пресненская набережная, 10блокС, Москва     Международная   
10102  Пресненский район, 123112, 1-й Красногвардейск...     Международная   

       Время до метро в минутах  Рейтинг  Отзывы  \
0                           9.0      5.0    37.0   
1                          13.0      4.8   141.0   
2                          21.0      5.0   290.0   
3                           9.0      4.9   156.0   
4                           8.0      4.8   477.0   
...                         ...      ...     ...   
10098                       4.0      4.2    12.0   
10099                       2.0      4.3     2.0   
10100                      10.0      2.8    13.0   
10101                       2.0      4.4    13.0   
10102                       5.0      5.0    16.0   

                                            Подкатегории        Город  Этаж  \
0      Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1      Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2      Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3      Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4      Онкологический диспансер, Перевозка больных, В...       Москва   NaN   
...                                                  ...          ...   ...   
10098  Регистрация юридических лиц и предпринимателей...       Москва   NaN   
10099  Юридическое сопровождение бизнеса, Юридическая...       Москва   NaN   
10100  Урегулирование договорных споров, Определение ...       Москва  10.0   
10101  Перевод в нежилое помещение, Юридическая консу...       Москва   NaN   
10102  Составление жалоб, Налоговые консультации, Адв...       Москва   NaN   

                Категория  ...  Число подписчиков Число постов за 6 месяцев  \
0      Медицинские центры  ...               1598                        99   
1      Медицинские центры  ...                511                        85   
2      Медицинские центры  ...                  0                         0   
3      Медицинские центры  ...                 16                         0   
4      Медицинские центры  ...                  0                         0   
...                   ...  ...                ...                       ...   
10098  Юридические услуги  ...                  0                         0   
10099  Юридические услуги  ...               3470                        62   
10100  Юридические услуги  ...                  0                         0   
10101  Юридические услуги  ...               2979                       100   
10102  Юридические услуги  ...                 73                        31   

       Среднее число комментариев на пост(%)  Среднее число лайко

In [29]:
final['Процент вовлеченности (ER)'] = final['Процент вовлеченности (ER)'] / 4
final

Название  \
0      Оздоровительный центр на Соколе   
1       Андреевские больницы - НЕБОЛИТ   
2                              WeiDent   
3              Клиника на Потаповскомъ   
4                  Юсуповская больница   
...                                ...   
10098                  Licenziataxi.ru   
10099                           Фемида   
10100  Листратенко Татьяна Анатольевна   
10101                            Базис   
10102               Петухов и партнеры   

                                             Адрес места             Метро  \
0      Балтийская улица, 15, 432 кабинет; 4 этаж, Москва             Сокол   
1                       Спасская улица, 1к3, Красногорск          Мякинино   
2                  улица Академика Опарина, 4 к1, Москва           Беляево   
3        Потаповский переулок, 10, вход со двора, Москва      Чистые пруды   
4                    Нагорная улица, 17 корпус 6, Москва          Нагорная   
...                                                  ...               ...   
10098     проезд Черепановых, 3, 10 офис; 2 этаж, Москва           Коптево   
10099  площадь Журавлева, 10 ст1, 120 офис; 1 этаж, М...  Электрозаводская   
10100           Люблинская улица, 179/1, 10 этаж, Москва           Марьино   
10101            Пресненская набережная, 10блокС, Москва     Международная   
10102  Пресненский район, 123112, 1-й Красногвардейск...     Международная   

       Время до метро в минутах  Рейтинг  Отзывы  \
0                           9.0      5.0    37.0   
1                          13.0      4.8   141.0   
2                          21.0      5.0   290.0   
3                           9.0      4.9   156.0   
4                           8.0      4.8   477.0   
...                         ...      ...     ...   
10098                       4.0      4.2    12.0   
10099                       2.0      4.3     2.0   
10100                      10.0      2.8    13.0   
10101                       2.0      4.4    13.0   
10102                       5.0      5.0    16.0   

                                            Подкатегории        Город  Этаж  \
0      Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1      Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2      Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3      Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4      Онкологический диспансер, Перевозка больных, В...       Москва   NaN   
...                                                  ...          ...   ...   
10098  Регистрация юридических лиц и предпринимателей...       Москва   NaN   
10099  Юридическое сопровождение бизнеса, Юридическая...       Москва   NaN   
10100  Урегулирование договорных споров, Определение ...       Москва  10.0   
10101  Перевод в нежилое помещение, Юридическая консу...       Москва   NaN   
10102  Составление жалоб, Налоговые консультации, Адв...       Москва   NaN   

                Категория  ...  Число подписчиков Число постов за 6 месяцев  \
0      Медицинские центры  ...               1598                        99   
1      Медицинские центры  ...                511                        85   
2      Медицинские центры  ...                  0                         0   
3      Медицинские центры  ...                 16                         0   
4      Медицинские центры  ...                  0                         0   
...                   ...  ...                ...                       ...   
10098  Юридические услуги  ...                  0                         0   
10099  Юридические услуги  ...               3470                        62   
10100  Юридические услуги  ...                  0                         0   
10101  Юридические услуги  ...               2979                       100   
10102  Юридические услуги  ...                 73                        31   

       Среднее число комментариев на пост(%)  Среднее число лайко

In [30]:
updated_file_path_final = 'zoon_data_final_.csv'
final.to_csv(updated_file_path_final, index=False, encoding='UTF-8-SIG')

In [52]:
metro = pd.read_excel('Метро.xlsx')
metro.head()

Метро                                    Район
0         Авиамоторная         Восточный Административный Округ
1        Автозаводская             Южный Административный Округ
2        Академическая      Юго-западный Административный Округ
3  Александровский сад       Центральный Административный Округ
4         Алексеевская  Северо-восточный Административный Округ

In [ ]:
left_join_df = pd.merge(df1, df2, on='Customer_id', how='left')

In [62]:
merged = pd.merge(final, metro, on='Метро', how='left')
merged

Название  \
0      Оздоровительный центр на Соколе   
1       Андреевские больницы - НЕБОЛИТ   
2                              WeiDent   
3              Клиника на Потаповскомъ   
4                  Юсуповская больница   
...                                ...   
10098                  Licenziataxi.ru   
10099                           Фемида   
10100  Листратенко Татьяна Анатольевна   
10101                            Базис   
10102               Петухов и партнеры   

                                             Адрес места             Метро  \
0      Балтийская улица, 15, 432 кабинет; 4 этаж, Москва             Сокол   
1                       Спасская улица, 1к3, Красногорск          Мякинино   
2                  улица Академика Опарина, 4 к1, Москва           Беляево   
3        Потаповский переулок, 10, вход со двора, Москва      Чистые пруды   
4                    Нагорная улица, 17 корпус 6, Москва          Нагорная   
...                                                  ...               ...   
10098     проезд Черепановых, 3, 10 офис; 2 этаж, Москва           Коптево   
10099  площадь Журавлева, 10 ст1, 120 офис; 1 этаж, М...  Электрозаводская   
10100           Люблинская улица, 179/1, 10 этаж, Москва           Марьино   
10101            Пресненская набережная, 10блокС, Москва     Международная   
10102  Пресненский район, 123112, 1-й Красногвардейск...     Международная   

       Время до метро в минутах  Рейтинг  Отзывы  \
0                           9.0      5.0    37.0   
1                          13.0      4.8   141.0   
2                          21.0      5.0   290.0   
3                           9.0      4.9   156.0   
4                           8.0      4.8   477.0   
...                         ...      ...     ...   
10098                       4.0      4.2    12.0   
10099                       2.0      4.3     2.0   
10100                      10.0      2.8    13.0   
10101                       2.0      4.4    13.0   
10102                       5.0      5.0    16.0   

                                            Подкатегории        Город  Этаж  \
0      Расслабляющий массаж, Массаж спины, Лимфодрена...       Москва   NaN   
1      Фракционный фототермолиз, Инъекции коллагена, ...  Красногорск   NaN   
2      Стоматология под микроскопом, Базальная имплан...       Москва   NaN   
3      Фракционный фототермолиз, Онкологический диспа...       Москва   NaN   
4      Онкологический диспансер, Перевозка больных, В...       Москва   NaN   
...                                                  ...          ...   ...   
10098  Регистрация юридических лиц и предпринимателей...       Москва   NaN   
10099  Юридическое сопровождение бизнеса, Юридическая...       Москва   NaN   
10100  Урегулирование договорных споров, Определение ...       Москва  10.0   
10101  Перевод в нежилое помещение, Юридическая консу...       Москва   NaN   
10102  Составление жалоб, Налоговые консультации, Адв...       Москва   NaN   

                Категория  ...  Число постов за 6 месяцев  \
0      Медицинские центры  ...                         99   
1      Медицинские центры  ...                         85   
2      Медицинские центры  ...                          0   
3      Медицинские центры  ...                          0   
4      Медицинские центры  ...                          0   
...                   ...  ...                        ...   
10098  Юридические услуги  ...                          0   
10099  Юридические услуги  ...                         62   
10100  Юридические услуги  ...                          0   
10101  Юридические услуги  ...                        100   
10102  Юридические услуги  ...                         31   

      Среднее число комментариев на пост(%)  Среднее число лайков на пост  \
0                                       8.0                          8.49   
1                                       5.0                          2.13   
2                                       0.0       

In [54]:
unique_values = final['Метро'].unique()
unique_df = pd.DataFrame(unique_values, columns=['Метро'])
unique_df


Метро
0                      Сокол
1                   Мякинино
2                    Беляево
3               Чистые пруды
4                   Нагорная
..                       ...
255  Библиотека имени Ленина
256                 Терехово
257             Черкизовская
258                   Косино
259               Тютчевская

[260 rows x 1 columns]

In [34]:
metro

Метро                                    Район
0           Авиамоторная         Восточный Административный Округ
1          Автозаводская             Южный Административный Округ
2          Академическая      Юго-западный Административный Округ
3    Александровский сад       Центральный Административный Округ
4           Алексеевская  Северо-восточный Административный Округ
..                   ...                                      ...
245     Электрозаводская         Восточный Административный Округ
246        Юго-Восточная     Юго-восточный Административный Округ
247         Юго-Западная          Западный Административный Округ
248                Южная             Южный Административный Округ
249              Ясенево      Юго-западный Административный Округ

[250 rows x 2 columns]

In [58]:
merged = pd.merge(unique_df, metro, on='Метро', how='left')
merged.fillna("нет значений", inplace=True)

merged.head(7)

Метро                                Район
0            Сокол      Северный Административный Округ
1         Мякинино      Западный Административный Округ
2          Беляево  Юго-западный Административный Округ
3     Чистые пруды   Центральный Административный Округ
4         Нагорная         Южный Административный Округ
5       Нет данных                         нет значений
6  Бунинская аллея  Юго-западный Административный Округ

In [56]:
import numpy as np
merged[merged['Район'] == 'нет значений']['Метро']

5    Нет данных
Name: Метро, dtype: object

In [63]:
updated_file_path_final = 'zoon_data_final_with_metro.csv'
merged.to_csv(updated_file_path_final, index=False, encoding='UTF-8-SIG')